test/apps/iodemo/io_demo.cc

```diff
@@ -1981,20 +2001,26 @@ class DemoClient : public P2pDemoCommon {
 
         io_op_perf_info.resize(IO_OP_MAX + 1);
         for (int op = 0; op <= IO_OP_MAX; ++op) {
-            io_op_perf_info[op].min       = std::numeric_limits<long>::max();
-            io_op_perf_info[op].max       = 0;
-            io_op_perf_info[op].min_index = _server_info.size();
-            io_op_perf_info[op].total     = 0;
+            io_op_perf_info[op].min         = std::numeric_limits<long>::max();
+            io_op_perf_info[op].max         = 0;
+            io_op_perf_info[op].min_index   = _server_info.size();
+            io_op_perf_info[op].total       = 0;
+            io_op_perf_info[op].total_bytes = 0;
         }
 
         // Collect min/max among all connections
         for (size_t server_index = 0; server_index < _server_info.size();
              ++server_index) {
-            server_info_t& server_info = _server_info[server_index];
-            long total_completed       = 0;
+            server_info_t& server_info   = _server_info[server_index];
+            long total_completed         = 0;
+            size_t total_bytes_completed = 0;
             for (int op = 0; op <= IO_OP_MAX; ++op) {
+                size_t bytes_completed;
                 long delta_completed;
                 if (op != IO_OP_MAX) {
+                    assert(server_info.bytes_sent[op] ==
```

Comment:
do we really need to maintain bytes_sent or is it just for the assertion?

Reply:
yes, this is just for assertion to make sure we did calculations accurate.
the first implementation contains the bug and I can't realize what's goign wrong. So, adding `bytes_sent` helps me to find out. Also, I think it is good self-test for the future, because it is quite easy to break something.